client/my-sites/email/email-providers-comparison/index.jsx

```diff
@@ -281,10 +286,16 @@ class EmailProvidersComparison extends Component {
 	onGoogleConfirmNewUsers = () => {
 		const { comparisonContext, domain, gSuiteProduct, hasCartDomain, source } = this.props;
 		const { googleUsers } = this.state;
+		const validatedUsers = validateUsers( googleUsers );
 
 		const usersAreValid = areAllUsersValid( googleUsers );
 		const userCanAddEmail = hasCartDomain || canCurrentUserAddEmail( domain );
 
+		this.setState( {
+			validatedMailboxUuids: validatedUsers.map( ( user ) => user.uuid ),
+			googleUsers: validatedUsers,
+		} );
```

Comment:
This is likely a possible issue for the Titan logic as well: we're calling `setState()` with an input that comes from `state` (i.e. `googleUsers`), which means we could technically end up operating on stale state data.

We should consider writing this logic using the callback form of `setState()` (as per [the React docs](https://reactjs.org/docs/react-component.html#setstate)). That may be something we should tackle in a follow-up PR, as it will require a more extensive update to the overall logic of this function.

Reply:
That's something super useful to not forget. Good catch 🔥